src/tests/electron/tests/android-setup-prompt-grant-permissions.test.ts

```diff
@@ -0,0 +1,80 @@
+import { scanForAccessibilityIssues } from 'tests/electron/common/scan-for-accessibility-issues';
+import { AndroidSetupViewController } from 'tests/electron/common/view-controllers/android-setup-view-controller';
+import { AppController } from 'tests/electron/common/view-controllers/app-controller';
+import {
+    commonAdbConfigs,
+    delayAllCommands,
+    setupMockAdb,
+    simulateServiceLacksPermissions,
+} from '../../miscellaneous/mock-adb/setup-mock-adb';
+
+describe('Android setup - prompt-grant-permissions', () => {
+    const [cancelId, nextId] = [leftFooterButtonAutomationId, rightFooterButtonAutomationId];
+    const defaultDeviceConfig = commonAdbConfigs['single-device'];
+    let app: AppController;
+    let dialog: AndroidSetupViewController;
+
+    beforeEach(async () => {
+        await setupMockAdb(simulateServiceLacksPermissions(defaultDeviceConfig));
+        app = await createApplication({ suppressFirstTimeDialog: true });
+        dialog = await app.openAndroidSetupView('prompt-grant-permissions');
+    });
+
+    afterEach(async () => {
+        if (app != null) {
+            await app.stop();
+        }
+    });
+
+    it('initial component state is correct', async () => {
+        expect(await dialog.isEnabled(getAutomationIdSelector(cancelId))).toBe(true);
+        expect(await dialog.isEnabled(getAutomationIdSelector(nextId))).toBe(false);
+        expect(await dialog.isEnabled(getAutomationIdSelector(tryAgainAutomationId))).toBe(true);
+    });
+
+    it('goes to prompt-choose-device upon cancel', async () => {
+        await setupMockAdb(commonAdbConfigs['multiple-devices']);
+        await dialog.client.click(getAutomationIdSelector(cancelId));
+        await dialog.waitForDialogVisible('prompt-choose-device');
+    });
+
+    it('try again returns here if permissions are not granted', async () => {
+        await setupMockAdb(
+            delayAllCommands(5000, simulateServiceLacksPermissions(defaultDeviceConfig)),
+        );
+        await dialog.client.click(getAutomationIdSelector(tryAgainAutomationId));
+        await dialog.waitForDialogVisible('detect-permissions');
+        await dialog.waitForDialogVisible('prompt-grant-permissions');
+    });
+
+    it('try again moves on if permissions are granted; detect-permissions a11y test', async () => {
+        await setupMockAdb(commonAdbConfigs['slow-single-device']);
+        await dialog.client.click(getAutomationIdSelector(tryAgainAutomationId));
+        await dialog.waitForDialogVisible('detect-permissions');
+        await a11yTest(true);
+        await a11yTest(false);
+        await dialog.waitForDialogVisible('configuring-port-forwarding');
+    });
+
+    it.each([true, false])(
+        'should pass accessibility validation with highContrastMode=%s',
+        async highContrastMode => a11yTest(highContrastMode),
+    );
+
+    const a11yTest = async highContrastMode => {
```

Comment:
in a separate PR I will see if we can do this more generally across the existing e2e tests since it will be a common pattern for us